tests/ld-floxlib.bats

```diff
@@ -0,0 +1,85 @@
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test if LD_AUDIT and ld-floxlib.so works with flox.
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# bats file_tags=end2end,ld-floxlib
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup
+}
+
+# ---------------------------------------------------------------------------- #
+
+# Helpers for project based tests.
+
+project_setup() {
+  export PROJECT_DIR="${BATS_TEST_TMPDIR?}/project-${BATS_TEST_NUMBER?}"
+  export PROJECT_NAME="${PROJECT_DIR##*/}"
+  rm -rf "$PROJECT_DIR"
+  mkdir -p "$PROJECT_DIR"
+  cp ./harnesses/ld-floxlib/* "$PROJECT_DIR"
+  pushd "$PROJECT_DIR" > /dev/null || return
+  "$FLOX_BIN" init
+  sed -i \
+    's/from = { type = "github", owner = "NixOS", repo = "nixpkgs" }/from = { type = "github", owner = "NixOS", repo = "nixpkgs", rev = "e8039594435c68eb4f780f3e9bf3972a7399c4b1" }/' \
+    "$PROJECT_DIR/.flox/env/manifest.toml"
+}
+
+project_teardown() {
+  popd > /dev/null || return
+  rm -rf "${PROJECT_DIR?}"
+  unset PROJECT_DIR
+  unset PROJECT_NAME
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup
+  project_setup
+}
+teardown() {
+  project_teardown
+  common_test_teardown
+}
+
+# ---------------------------------------------------------------------------- #
+#
+@test "test ld-floxlib.so on Linux only" {
+  if [ $(uname -s) != "Linux" ]; then
+    skip "not Linux"
+  fi
+
+  run "$FLOX_BIN" install gcc glibc giflib
+  assert_success
+  assert_output --partial "✅ 'gcc' installed to environment"
+  assert_output --partial "✅ 'glibc' installed to environment"
+  assert_output --partial "✅ 'giflib' installed to environment"
+
+  #SHELL=bash run expect -d "$TESTS_DIR/ld-floxlib.exp" "$PROJECT_DIR"
+  #assert_success
+
+  ### Verify environment
+  run "$FLOX_BIN" activate -- sh ./verify-environment.sh
+  assert_success
+
+  ### Test 1: load libraries found in $FLOX_ENV_LIB_DIRS last
+  run "$FLOX_BIN" activate -- sh ./test-load-library-last.sh
+  assert_success
+
+  ### Test 2: confirm LD_AUDIT can find missing libraries
+  run "$FLOX_BIN" activate -- sh -c "cc -o print-gif-info ./print-gif-info.c -lgif && ./print-gif-info ./flox-edge.gif"
+  assert_output --partial "GIF Information for: ./flox-edge.gif
+  assert_output --partial "Number of frames: 0
+  assert_output --partial "Width: 270 pixels
+  assert_output --partial "Height: 137 pixels
+}
```

Comment:
suggestion: add a test that shows that with the ld loader disabled we don't find the lib?

Reply:
Sorry; yeah discussed with Matthew that this is still a WIP. Will push a patch which walks through all of the various possibilities using LD_AUDIT to load last and LD_LIBRARY_PATH to load first.

Reply:
@ysndr - I followed your suggestion and added the test to confirm it breaks with LD_AUDIT not defined and found on fedora it "fails" because fedora ships with libgif installed by default. 🤦 

What we want is a library that is guaranteed not to be found on the "system", but that is available to be installed as a package to an environment. The best way to do that would be to create our own custom package, but we can't do that yet, and we can't even use `nix-store --add` to create a symlink'd alias for a known library because we cannot install by `/nix/store` path anymore. So I'm a tad stuck.  :-\

Reply:
@ysndr I realized we could use Nix itself (linking against `libnixmain.so` to call `nix::printVersion()`). That's a library that is guaranteed not to be installed in any "system" FHS location and is available to be installed to a flox environment. The updated test is pushed in commit 079677fe22e02690985dc4d99a1185cb57e56754.

Reply:
I'll go ahead and mark this resolved as it is implemented in [ld-floxlib.bats, lines 110-116](https://github.com/flox/flox/blob/2c889a5e366233effca1ba352c44b9ecdf6dcddc/tests/ld-floxlib.bats#L110-L116).

Reply:
I'll go ahead and mark this resolved as it is implemented in [ld-floxlib.bats, lines 110-116](https://github.com/flox/flox/blob/2c889a5e366233effca1ba352c44b9ecdf6dcddc/tests/ld-floxlib.bats#L110-L116).